rotkehlchen/chain/aggregator.py

```diff
@@ -919,18 +919,12 @@ def query_evm_chain_balances(self, chain: SUPPORTED_EVM_CHAINS) -> None:
 
         # Query native token balances
         manager = cast('EvmManager', self.get_chain_manager(chain))
-        native_token_usd_price = Inquirer().find_usd_price(manager.node_inquirer.native_token)
         chain_balances = self.balances.get(chain)
-        queried_balances = manager.node_inquirer.get_multi_balance(accounts)
-        for account, balance in queried_balances.items():
-            if balance == ZERO:
-                continue
-
-            usd_value = balance * native_token_usd_price
+        for account, balance in manager.node_inquirer.get_multi_balance(accounts).items():
             chain_balances[account] = BalanceSheet(
                 assets=defaultdict(Balance, {
-                    manager.node_inquirer.native_token: Balance(balance, usd_value),
-                }),
+                    manager.node_inquirer.native_token: Balance(balance, balance * Inquirer().find_usd_price(manager.node_inquirer.native_token)),  # noqa: E501
```

Comment:
I would keep the price query outside the for loop for different reasons:

1. The asset is constant
2. Is one extra function call that has to context switch
3. You are calling the `Inquirer()` class that is creating the object and this is extra logic that you don't need to execute
4. It can happen that the price goes out of the cache and you make additional queries

I would keep as before and just do usd_value = ZERO if balance == ZERO